### ISSUE TITLE
Invalid npm package dependencies

### ISSUE THREAD

**Issue by konraddysput** [nestjs/terminus#233]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x ] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
When I download latest version of libraries via npm:

```
$ npm install --save @nestjs/terminus @godaddy/terminus

```
and I'm trying to use healtchecks like below: 

```
import {
  TerminusEndpoint,
  TerminusOptionsFactory,
  DNSHealthIndicator,
  TypeOrmHealthIndicator,
  TerminusModuleOptions,
  MemoryHealthIndicator,
  DiskHealthIndicator,
} from '@nestjs/terminus';
import { Injectable } from '@nestjs/common';

@Injectable()
export class HealthCheck implements TerminusOptionsFactory {
  constructor(
    private readonly _dns: DNSHealthIndicator,
    private readonly _typeorm: TypeOrmHealthIndicator,
    private readonly _memoryHealthIndicator: MemoryHealthIndicator,
    private readonly _diskHealthIndicator: DiskHealthIndicator,
  ) {}

  public createTerminusOptions(): TerminusModuleOptions {
    const healthEndpoint: TerminusEndpoint = {
      url: '/health',
      healthIndicators: [
        async () => this._dns.pingCheck('google', 'https://google.com'),
        async () => this._typeorm.pingCheck('database', { timeout: 1500 }),
        async () =>
          this._diskHealthIndicator.checkStorage('storage', { thresholdPercent: 0.5, path: '/' }),
        async () => this._memoryHealthIndicator.checkRSS('rss', 500 * 1024 * 1024),
        async () => this._memoryHealthIndicator.checkHeap('heap', 100 * 1024 * 1024),
        async () => this._typeorm.pingCheck('database', { timeout: 1500 }),
      ],
    };
    return {
      endpoints: [healthEndpoint],
    };
  }
}

```
I'm receiving build errors. Please check image below:
![image](https://user-images.githubusercontent.com/9386268/59285723-b86db700-8c6e-11e9-8281-2fdce55117aa.png)

I tried to install manually packages, but it still doesn't work.

## Expected behavior
After `$ npm install --save @nestjs/terminus @godaddy/terminus` I expect to have every needed dependency, or detailed instruction what I should do if I want to use providers from official doc:
https://docs.nestjs.com/recipes/terminus

## Minimal reproduction of the problem with instructions

```
$ npm install --save @nestjs/terminus @godaddy/terminus
$ tsc build
```

## Environment

"@godaddy/terminus": "^4.1.2",
"@nestjs/terminus": "^6.3.1",
NodeJS: v10.16.0
Windows 10

**Comment by BrunnerLivio** [comment]
I tried it out myself and it works with a clean setup:

```
nest new terminus-test
cd terminus-test
npm i -s @nestjs/terminus @godaddy/terminus

```

I have added 1:1 your code and received the following result, which is expected at the current state of 6.3.x:

```
{
  "status": "error",
  "error": {
    "rss": {
      "status": "up"
    },
    "heap": {
      "status": "up"
    },
    "storage": {
      "status": "up"
    },
    "google": {
      "status": "up"
    },
    "database": {
      "status": "down",
      "message": "Connection provider not found in application context"
    }
  },
  "details": {
    "rss": {
      "status": "up"
    },
    "heap": {
      "status": "up"
    },
    "storage": {
      "status": "up"
    },
    "google": {
      "status": "up"
    },
    "database": {
      "status": "down",
      "message": "Connection provider not found in application context"
    }
  }
}
```

**Comment by konraddysput** [comment]
@BrunnerLivio are you using typescript?

**Comment by BrunnerLivio** [comment]
@konraddysput yes

```
➜ tsc -v
Version 3.2.2

➜ node -v   
v12.4.0
```

**Comment by konraddysput** [comment]
You have right - I started project from the beginning like you did and I found what was a problem.  My tsconfig: 

```
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": { "*": ["types/*"] },
    "target": "es2017",
    "module": "commonjs",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "./dist/",
    "incremental": true,
    "strict": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noUnusedLocals": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "allowSyntheticDefaultImports": true
  },
  "exclude": ["./tests", "./dist", "./node_modules", "dist", "**/*spec.ts"]
}

```
Do you see anything wrong here?

**Comment by konraddysput** [comment]
OK, I see a problem is with two typescript flags:

```
"strict": true,
"noImplicitAny": true,
```

@BrunnerLivio do you know why I cannot use them with your library? Actually I need to use them in my code.

**Comment by BrunnerLivio** [comment]
@konraddysput sorry for the late reply. I was able to reproduce your error. I worked around this problem by using my [custom types](https://github.com/nestjs/terminus/blob/master/types/check-disk-space.d.ts). This is not sustainable, so I will create a PR to the library itself to support the typings in strict mode. You can follow the discussion here: Alex-D/check-disk-space#3


If you want to workaround this, you have to set the setting in your `tsconfig.json`
```json
{
  "compilerOptions": {
    "paths": { "*": ["types/*"] }
  }
}
```

and add the types folder with `check-disk-space.d.ts`

```bash
mkdir types
cat <<EOT >> types/check-disk-space.d.ts
import { totalmem } from 'os';

export = index;
declare function index(
  directoryPath: string,
): Promise<{ free: number; size: number }>;
declare namespace index {
  class InvalidPathError {
    constructor(message: any);
    name: any;
    message: any;
  }
  class NoMatchError {
    constructor(message: any);
    name: any;
    message: any;
  }
  function getFirstExistingParentPath(directoryPath: any): any;
}

EOT
```

**Comment by konraddysput** [comment]
Thanks @BrunnerLivio :) I will wait for new version then.

**Comment by konraddysput** [comment]
Hey @BrunnerLivio 

Today I had a chance to update package add check if types works with strict mode. Unfortunately, TypeScript still complain about missing types. 

I had a chance to add `check-disk-space` types definition to project, but it still doesn't work. Now I have problem with `mongoose.health.d.ts`

Thanks!

**Comment by BrunnerLivio** [comment]
I think it would make sense if you would use `skipLibCheck` - or is anything against this option?

```typescript
{
  "compilerOptions": {
    "skipLibCheck": true
    ...
  }
}

```

**Comment by BrunnerLivio** [comment]
Screw `skipLibCheck` - fixed with `6.5.0`
